### ISSUE TITLE
when calculating test loss

### ISSUE THREAD

**Issue by algebra-cadabra** [mari-linhares/tensorflow-maml#4]
Hello thanx form the code it helped alot!

i have some questions for calculating test loss in MAML

when your code calculates test loss in train_maml() for specific sine function in inner loop

are you using same data to calculate both training and test loss? 
shouldn't you sample new test data from same sine function to calculate test error like in the paper

thanx

**Comment by algebra-cadabra** [comment]
I think you are optimizing for dropping training errors in meta learning task whereas in the paper it says you should optimize for dropping test errors in Algorithm 2 / line 2 in the paper.

Thanx

**Comment by shufflebyte** [comment]
you are right. you should sample new sine waves (not only 10 new samples of the same wave) for validation. If you take a look into the code of Finn et al. you can see, that they sample from new sines too.

**Comment by hereismari** [comment]
Yes, thanks for the catch! Feel free to send a PR!

**Comment by algebra-cadabra** [comment]
Thanx to both of you ~~!
